pkg/syncer/cnsoperator/controller/cnsnodevmattachment/cnsnodevmattachment_controller.go

```diff
@@ -416,27 +417,25 @@ func (r *ReconcileCnsNodeVMAttachment) Reconcile(ctx context.Context,
 					msg = fmt.Sprintf("failed to verify is VM CR is present with UUID: %s "+
 						"in namespace: %s", nodeUUID, request.Namespace)
 					recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
-					return reconcile.Result{RequeueAfter: timeout}, nil
+					return reconcile.Result{RequeueAfter: timeout}, "NotComputed", nil
 				}
 				if !isPresent {
 					msg = fmt.Sprintf("VM CR is not present with UUID: %s in namespace: %s. "+
 						"Removing finalizer on CnsNodeVMAttachment: %s instance.",
 						nodeUUID, request.Namespace, request.Name)
-					// This check is needed in scenarios where VC inventory is stale due to upgrade or back-up and restore
-					removeFinalizerFromCRDInstance(ctx, instance, request)
-					err = updateCnsNodeVMAttachment(ctx, r.client, instance)
-					if err != nil {
-						log.Errorf("updateCnsNodeVMAttachment failed. err: %v", err)
-					}
-					recordEvent(ctx, r, instance, v1.EventTypeNormal, msg)
-					return reconcile.Result{}, nil
 				} else {
-					msg := fmt.Sprintf("VM on VC not found but VM CR with UUID: %s "+
-						"is still present in namespace: %s. Retrying the operation",
-						nodeUUID, request.Namespace)
-					recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
-					return reconcile.Result{RequeueAfter: timeout}, nil
+					msg = fmt.Sprintf("VM on VC not found but VM CR with UUID: %s "+
+						"is still present in namespace: %s. Returning success since "+
+						"the deletion timestamp is set.", nodeUUID, request.Namespace)
```

Comment:
This flow gets executed when VM is not present, VM CR is present and the deletion timestamp is set on CnsNodeVmAttachment CR. However, it's not checking the deletion timestamp on VM CR, if you're trying to address this comment - https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/1744#issuecomment-1125429674

Reply:
This block is actually inside `if instance.DeletionTimestamp != nil {` in Line 396. The diff doesn't show this correctly, I think.

Reply:
Looks like I misread it NodeVMAttachment CR. Latest commit should have addresses the concern. PTAL